### PR TITLE
Implement market page markup and functionality to observe available shares 

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR implements market page markup and functionality to observe available shares on the market.
View of the market page:
Screen Recording 2023-06-06 at 17 24 14

### CODE REVIEW

**Review by Vladyslav-Kuksiuk — COMMENTED** [review]
LGTM.

File: client/src/main/kotlin/io/spine/examples/shareaware/client/Application.kt [hunk 1]
```diff
@@ -75,7 +76,7 @@ public fun application(): Unit = application {
                 when (currentPage.value) {
                     Page.HOME -> Text("HOME")
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/shareaware/client/Application.kt:77]: Let's make all cases lead to something called `...Page`. Not just render `Text(...)`.

File: client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+/**
+ * UI model for the `MarketPage`.
+ */
+public class MarketPageModel(client: DesktopClient) {
+    private val sharesSubscriptions: EntitySubscription<AvailableMarketShares> =
+        EntitySubscription(AvailableMarketShares::class.java, client, MarketProcess.ID)
+
+    /**
+     * Returns the current state of available shares on the market.
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt:76]: Please comment on its nullability.

**Review by armiol — APPROVED** [review]
@Artem-Semenov-dev LGTM with a couple of comments to address.

Also, after this PR, please have a look at the UI guidelines (Material 3, [Mac UI Guidelines](https://developer.apple.com/design/human-interface-guidelines/designing-for-macos) are good places to start), and apps that are distributed via markets (such as Mac Store). We need to make the UI much more orthodox in terms of look&feel.

As far as my googling goes, here is one of the similar apps, which we could use as a look&feel reference:

https://support.apple.com/guide/stocks/view-stocks-st96acfa8842/mac

I am sure there are many more.

**Pull request merged**